### DISCUSSION TITLE
ScriptReference/Unity.Collections.Allocator.TempJob

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#6] [Notes]
# ScriptReference/Unity.Collections.Allocator.TempJob

https://docs.unity3d.com/ScriptReference/Unity.Collections.Allocator.TempJob.html

**Comment by vertxxyz** [comment]
**`Allocator.TempJob`** is a slower allocation than [`Temp`](https://docs.unity3d.com/ScriptReference/Unity.Collections.Allocator.Temp.html) but is faster than [`Persistent`](https://docs.unity3d.com/ScriptReference/Unity.Collections.Allocator.Persistent.html). Use it for thread-safe allocations within a lifespan of four frames.
**Important**: You must `Dispose` of this type of allocation within four frames, or the console prints a warning, generated from the native code. Most small jobs use this `NativeContainer` allocation type.
(See [NativeContainer](https://docs.unity3d.com/Manual/JobSystemNativeContainer.html).)
